From f235d7344bda8fb8e554b278f85c0727617e037a Mon Sep 17 00:00:00 2001
From: Freddie Jheng <Freddie.Jheng@quantatw.com>
Date: Wed, 7 Jan 2026 16:30:21 +0800
Subject: [PATCH 2/4] configuration: santabarbara: add MCTP I2C target for SWB
 die temp

Add SWB_PEX as an MCTP I2C target on bus 94/102 with address 0x62
to enable die temperature monitoring via PLDM sensor.

Due to I3C hub initialization, the device may appear on either bus 94
or bus 102. Use different names to create separate inventory objects,
allowing mctpreactor to attempt connection on both buses and succeed
on whichever bus the device appears.

Change-Id: I45283deecda1fca687c91b557665098dff0ffcfa
Signed-off-by: Freddie Jheng <Freddie.Jheng@quantatw.com>
---
 .../santabarbara_pcie_switch_board_brcm.json         | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/configurations/meta/santabarbara/santabarbara_pcie_switch_board_brcm.json b/configurations/meta/santabarbara/santabarbara_pcie_switch_board_brcm.json
index f7580ba..f654a4d 100644
--- a/configurations/meta/santabarbara/santabarbara_pcie_switch_board_brcm.json
+++ b/configurations/meta/santabarbara/santabarbara_pcie_switch_board_brcm.json
@@ -425,6 +425,18 @@
             "Name": "Santabarbara_SWB_CPLD",
             "Type": "LatticeLCMXO3D_9400Firmware"
         },
+        {
+            "Address": "0x62",
+            "Bus": "114",
+            "Name": "SWB_PEX",
+            "PLDMSensors": [
+                {
+                    "ID": 1,
+                    "Name": "TEMP_C"
+                }
+            ],
+            "Type": "MCTPI2CTarget"
+        },
         {
             "Address": "0x42",
             "Bus": "33",
-- 
2.47.3

